docs/src/query_interface.md

```diff
@@ -0,0 +1,120 @@
+# Query Interface
+
+```@meta
+DocTestSetup =  quote
+    using QuantumSavory
+end
+```
```

Comment:
There should be an introductory paragraph or two explaining what is the point of all of this and why people should care about it. What problems does it solve (mostly, composability of protocols where one protocol can simply rely on queries and tags, instead of having to know the internals of other protocols).